lib/matplotlib/ticker.py

```diff
@@ -1212,12 +1216,32 @@ class EngFormatter(Formatter):
          24: "Y"
     }
 
-    def __init__(self, unit="", places=None):
+    def __init__(self, unit="", places=None, sep=" "):
+        """
+        Parameters
+        ----------
+        unit: string (default: "")
+            Unit symbol to use.
+
+        places: int (default: None)
+            Precision, i.e. number of digits after the decimal point.
+            If it is None, falls back to the floating point format '%g'.
+
+        sep: string (default: " ")
+            String used between the value and the prefix/unit. Beside the
+            default behavior, some other useful use cases may be:
+            * sep="" to append directly the prefix/unit to the value;
+            * sep="\\u00a0" to use a no-break space;
```

Comment:
I can remove this line if you think it is not that useful. I added the information just because (at least) in French typography, one are supposed to use a no-break space before units. And sometimes, I directly use an EngFormatter instance to format a value and its unit in a longer string (that is not a tick label). So I guess there might be some edges cases where unfortunate wrapping could happen.